backend/layers/processing/dataset_metadata_update.py

```diff
@@ -0,0 +1,190 @@
+                seurat.Misc(object=rds_object, slot=key)[0] = val
+
+        base.saveRDS(rds_object, file=seurat_filename)
+
+        self.create_artifact(
+            seurat_filename,
+            DatasetArtifactType.RDS,
+            key_prefix,
+            new_dataset_version_id,
+            self.artifact_bucket,
+            DatasetStatusKey.RDS,
+            datasets_bucket=self.datasets_bucket,
+        )
+        self.update_processing_status(new_dataset_version_id, DatasetStatusKey.RDS, DatasetConversionStatus.CONVERTED)
+
+    def update_cxg(
+        self,
+        cxg_s3_uri: str,
+        key_prefix: str,
+        dataset_version_id: DatasetVersionId,
+    ):
+        # TODO: update cxg metadata
+        new_cxg_dir = f"s3://{self.cellxgene_bucket}/{key_prefix}.cxg/"
+        self.s3_provider.upload_directory(cxg_s3_uri, new_cxg_dir)
+        self.update_processing_status(dataset_version_id, DatasetStatusKey.CXG, DatasetConversionStatus.CONVERTED)
+
+    def update_metadata(
+        self,
+        collection_version_id: CollectionVersionId,
+        dataset_version_id: DatasetVersionId,
+        metadata_update_dict: Dict[str, str],
+    ):
+        original_dataset_version = self.business_logic.get_dataset_version(dataset_version_id)
+        if original_dataset_version.status.processing_status != DatasetProcessingStatus.SUCCESS:
+            self.logger.info(f"Dataset {dataset_version_id} is not successfully processed. Skipping metadata update.")
+            return
+
+        artifact_uris = {artifact.type: artifact.uri for artifact in original_dataset_version.artifacts}
+        raw_h5ad_uri = artifact_uris[DatasetArtifactType.RAW_H5AD]
+
+        new_dataset_version_id, dataset_id = self.business_logic.ingest_dataset(
+            collection_version_id=collection_version_id,
+            url=raw_h5ad_uri,
+            file_size=0,
+            existing_dataset_version_id=dataset_version_id,
+            start_step_function=False,
+        )
+
+        self.process(new_dataset_version_id, raw_h5ad_uri, self.artifact_bucket)
+
+        key_prefix = self.get_key_prefix(new_dataset_version_id)
+
+        if DatasetArtifactType.H5AD in artifact_uris:
+            self.update_h5ad(
+                artifact_uris[DatasetArtifactType.H5AD],
+                key_prefix,
+                original_dataset_version,
+                new_dataset_version_id,
+                metadata_update_dict,
+            )
+
+        if DatasetArtifactType.RDS in artifact_uris:
+            self.update_rds(
+                artifact_uris[DatasetArtifactType.RDS], key_prefix, new_dataset_version_id, metadata_update_dict
+            )
+
+        if DatasetArtifactType.CXG in artifact_uris:
+            self.update_cxg(artifact_uris[DatasetArtifactType.CXG], key_prefix, new_dataset_version_id)
+
+        self.update_processing_status(new_dataset_version_id, DatasetStatusKey.H5AD, DatasetConversionStatus.CONVERTED)
```

Comment:
why isn't this in update_h5ad

Reply:
I had interpreted this as 'we've now converted H5AD to the other two formats' but I code-searched and you're right that we use this to mean we've finished labeling the H5AD. will fix